arch/x86_64/mcount.S

```diff
@@ -1,8 +1,54 @@
 /* argument passing: %rdi, %rsi, %rdx, %rcx, %r8, %r9 */
 /* return value: %rax */
 /* callee saved: %rbx, %rbp, %rsp, %r12-r15 */
-/* stack frame (with -pg): return addr = 8(%rbp), prev fp = 0(%rbp) */
-/* stack frame (with -fentry): return addr = (%rsp), prev fp = %rbp */
+/* stack frame (with -pg): parent addr = 8(%rbp), child addr = 56(%rsp) */
```

Comment:
No, using push/pop or manual adjusting is essentially same.  What matters is the offset it changes.